scripts/generate-meetup-data.js

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+var _ = require('lodash');
+var fs = require('fs');
+var targetFileName = 'free-code-camp-events.mock.json';
+var targetPath = 'js/';
+var meetup_api_key = _.get(process, 'env.MEETUP_API_KEY');
+var targetGroupUrlname = 'Free-Code-Camp-Indy';
+var targetStatus = ['upcoming', 'past'];
+
+function generateNewMeetupMockData(meetupApiKey) {
+  var meetup = require('meetup-api')({key: meetupApiKey});
+  return new Promise(function (resolve, reject) {
+    meetup.getEvents({group_urlname: targetGroupUrlname, status: targetStatus}, function (err, response) {
+      if (err) {
+        reject(new Error(err));
+      } else {
+        // Strip property that has secrets information!
+        var results = {
+          results: _.get(response, 'results')
+        }
+        resolve(results);
+      }
+    });
+  });
+};
+function writeNewData(mockJson) {
+  return new Promise(function (resolve, reject) {
+    fs.writeFile(targetPath + targetFileName, mockJson, function (err) {
+      if (err) {
+        reject(new Error(err))
+      } else {
+        resolve('Successfully written to file!');
+      }
+    });
+  });
+};
+function handleError(error) {
+  console.error(error);
+  process.exit(1);
+}
+
+// "main"
+if (_.isUndefined(meetup_api_key)) {
+  var error = new Error('MEETUP_API_KEY environment variable not found.  Please reference the README.md for instructions on how to set this.  Ending script.');
+  handleError(error);
+} else {
+  // TODO: Refactor this "callback arrow" antipattern to perhaps use ESNext's `async`/`await` functionality.
+  var getMockJson = generateNewMeetupMockData(meetup_api_key);
+  getMockJson.then(function (mockJson) {
+    var formattedMockJson = JSON.stringify(mockJson, null, 2);
+    var writeResult = writeNewData(formattedMockJson);
+    writeResult.then(function () {
+      console.log('Successfully wrote new data to target file');
+      process.exit(0);
+    }, handleError);
+  }, handleError);
+}
```

Comment:
RE: "perhaps use ESNext's `async`/`await` functionality"

In order to accomplish this we must either hook in a "transpile" step of to make ESNext runnable on ES5 machines.  We could alternatively require developers to use node >= 6 or utilize "node version manager" `nvm`.  Doing this would allow developers to write ESNext on all files not run on the client (aka non web asset files).

To add the transpile step we could set up the project to have both a `src` and a `dist` folder; `dist` representing the compiled ES5 versions of code. `src/scripts/generate-meetup-data.es6.js` could then be transpiled into `dist/scripts/generate-meetup-data.js`.  Then you must update `package.json` to reference the compiled ES5 code (`"generate": "node dist/scripts/generate-meetup-data.js"`).  We already have `gulp` so adding a process wouldn't be a huge step.

Once you _can_ write ESNext, the above logic could be simplified as:
```
(async function main () {
  const mockJson = await generateNewMeetupMockData(meetup_api_key);
  const formattedMockJson = JSON.stringify(mockJson, null, 2);
  const writeResult = await writeNewData(formattedMockJson);
  console.log('Successfully wrote new data to target file');
  process.exit(0);
})()
```
... rewriting the error handling in a `async`/`await` friendly way.